src/semantics/mod.rs

```diff
@@ -356,7 +356,7 @@ where
             let raw_ty_fields = raw_ty_fields.unnest();
 
             if raw_fields.len() != raw_ty_fields.len() {
-                unimplemented!();
+                return Err(TypeError::Internal(InternalError::Unimplemented{feat: "record size mismatch. (change to RecordSizeMismatch)".to_string()}));
```

Comment:
Do you have rustfmt running? I try to have this on by default. Also we can probably remove this whole check because it is already covered in the check you added in #130! 😄

Reply:
I made that return an error *just* in case. It should never happen with the previous check, but you can never be too careful.